ibis/formats/pandas.py

```diff
@@ -150,12 +150,15 @@ def convert_table(cls, df, schema):
     def convert_column(cls, obj, dtype):
         pandas_type = PandasType.from_ibis(dtype)
 
-        if obj.dtype == pandas_type and dtype.is_primitive():
-            return obj
-
         method_name = f"convert_{dtype.__class__.__name__}"
         convert_method = getattr(cls, method_name, cls.convert_default)
 
+        if convert_method is cls.convert_default:
+            # only take the fast path if it isn't overridden in a subclass
```

Comment:
Then the fast path itself should be implemented in `convert_default()` instead.

Reply:
Done.